### ISSUE TITLE
Hi10p is not always respected

### ISSUE THREAD

**Issue by Draic** [MediaBrowser/plugin.video.emby#145]
With "Transcode hi10p" enabled the video does not get transcoded.

h264 10bit profile files cannot be played on my device, so I rely on transcoding in this case. The hi10p option does not trigger the transcoding and playback will fail. Only force transcode works.

After looking into the playutils.py, I believe this is due to checking only the profile for the phrase "high 10". Instead it should check for the actual bitdepth (10 bit), as not every 10bit encoding will use the high 10 profile and some use High 4:4:4 instead. The bitdepth is 10bit in both cases and both require transcode on a lot of devices. 

tl;dr
currently the hi10p option does not catch all cases of 10bit encoding. it should also check for the bitdepth instead of profile alone

**Comment by Draic** [comment]
So just tested out version 3.1.10a: Indeed the hi10p recognition has been improved and works as expected. All 10 bit test cases I tried were auto triggering the transcoder, so seems this issue has been resolved.

But I discovered a bug with the transcoder on that version: Atleast when starting a file that has audio in flac stereo the transcoding will not process it. The video gets send to kodi with no audio (info screen shows "audio: unknown. 0hz"). Other sample files with different audio formats work fine with transcoding. This issue was not present on the stable 3.0.34, where flac audio gets transcoded as it should.

**Comment by angelblue05** [comment]
Are you able to provide your Kodi log that captures this issue?

**Comment by Draic** [comment]
sure. 
Wasn't sure if you needed debug loglevel so I recorded the same twice, with and without.
What I did: Start kodi, wait for emby to connect, start the video with transcoding, let it playback for a moment, stopped and exited kodi. Logs are from a different PC than my TV-System, but same issue on both.

[kodidebug.log](https://github.com/MediaBrowser/plugin.video.emby/files/2367378/kodidebug.log)
[kodinodebug.log](https://github.com/MediaBrowser/plugin.video.emby/files/2367380/kodinodebug.log)

**Comment by angelblue05** [comment]
fyi always without Kodi debug unless it's a special case and I ask you for it. Looking at your logs and will keep posted on the solution.

**Comment by angelblue05** [comment]
@Draic 
Can you provide an example of the transcode stream with 3.0.34? Just so I can see where the difference is. With 3.6x, the server provides the url so I'm not sure why it would have difficulty with flac audio.

**Comment by Draic** [comment]
of course. 
All I can tell you is that the problem is consistent. Flac will not work on any video that I tried, but with the old addon version it is working just fine. From random testing I couldn't see a second format that the addon had an issue with, just flac.
[kodi.log](https://github.com/MediaBrowser/plugin.video.emby/files/2391497/kodi.log)

**Comment by angelblue05** [comment]
Thanks for that. Looks like we need to convert flac to something else for video playback. At least, according to your log, it would convert to ac3

**Comment by Draic** [comment]
hmm maybe it is a limitation (or bug) with the emby transcoder then? The hardware that is providing my logs is capable of  flac playback. If I do not force transcode on here the video and audio will work. Info shows audiochannel: flac. That is different during transcoding with the beta addon. Here atleast kodi does not recognize that this is a flac audio channel and shows an empty audio stream. 

But I guess it does not matter much. I am totally fine with converting to ac3. I doubt that there would be many cases in which you have the need to transcode for video, but are fine with flac audio.

**Comment by angelblue05** [comment]
Should be resolved with 3.1.20. I luckily had a file with audio flac and managed to reproduce this issue. Thanks for your patience.

**Comment by Draic** [comment]
Glad I could help
